lib/saputils.pm

```diff
@@ -31,63 +32,144 @@ SAP Applications.
-    Parses this input, returns a hash of hashes containing values for each host.
-
-    Output like:
+    Expect `SAPHanaSR-showAttr --format=$input_format` as input.
+    Returns pasrsed perl value decoded from json remap output from like :
             Hosts/vmhana01/remoteHost="vmhana02"
             Hosts/vmhana01/sync_state="PRIM"
             Hosts/vmhana01/vhost="vmhana01"
             Hosts/vmhana02/remoteHost="vmhana01"
             Hosts/vmhana02/sync_state="SOK"
             Hosts/vmhana02/vhost="vmhana02"
-    result in
+
+    output look a like:
     {
-        vmhana01 => {
-            remoteHost => 'vmhana02',
-            sync_state => 'PRIM',
-            vhost => 'vmhana01',
-        },
-        vmhana02 => {
-            remoteHost => 'vmhana01',
-            sync_state => 'SOK',
-            vhost => 'vmhana02',
-        },
-    }
+          'Host' => {
+                      'vmhana01' => {
+                                           'clone_state' => 'DEMOTED',
+                                           'score' => '100',
+                                           'srah' => '-',
+                                           'version' => '2.00.077.00.1710325774',
+                                           'vhost' => 'vmhana01',
+                                           'site' => 'site_a'
+                                         },
+                      'vmhana02' => {
+                                           'clone_state' => 'PROMOTED',
+                                           'score' => '150',
+                                           'vhost' => 'vmhana02',
+                                           'srah' => '-',
+                                           'version' => '2.00.077.00.1710325774',
+                                           'site' => 'site_b'
+                                         }
+                    },
+          'Site' => {
+                      'site_a' => {
+                                     'srMode' => 'sync',
+                                     'srPoll' => 'SOK',
+                                     'mns' => 'vmhana01'
+                                   },
+                      'site_b' => {
+                                     'srPoll' => 'PRIM',
+                                     'mns' => 'vmhana02',
+                                     'srMode' => 'sync'
+                                   }
+                    }
+        };
+
+
 
 =over 1
 
-=item B<input> - stdout of 'SAPHanaSR-showAttr --format=script'
+=item B<input_format> - format of the 'SAPHanaSR-showAttr --format='
+
+=item B<input> - stdout of 'SAPHanaSR-showAttr --format=<input_format>'
 
 =back
 =cut
 
 
 sub calculate_hana_topology {
     my (%args) = @_;
-    croak("Argument <input> missing") unless $args{input};
+    croak("calculate_hana_topology [ERROR] Argument <input> missing") unless $args{input};
+    my $input_format = $args{input_format} || 'script';
```

Comment:
Does only support `json` and `script` right? Maybe we can add a `croak` to check  input is one or the other.

Reply:
[SOLVED] croak added https://github.com/os-autoinst/os-autoinst-distri-opensuse/pull/21264/files#diff-4a40a74cf6048efd08f8d55e2613d752fb683e6dc56fc331da7e8f23ac0af1e9R102